### PR TITLE
Fix auth

### LINKED ISSUE #1799: Sunset Auth0 Lock
## Issue - Enhancement

This describes the final bit of clean up involved in migrating to the new cross-origin auth pages. We'll just need to remove the `newSignIn` release flag (as if it were always true), and remove the old landing page, including the auth0 lock.

### Acceptance Criteria (optional)
- Default experience stays the same; you still see the new auth when you reach the home page
- No more LandingPage or Auth0Lock
- auth0 lock removed from the package dependencies
- `newSignIn` release flag removed, and all the toggle points "flattened out" such that the new code paths are statically live, while the old ones are simply removed

- **Estimated effort:** 11 points ([see CONTRIBUTING.md](https://github.com/ParabolInc/action/blob/master/CONTRIBUTING.md#points-and-sizes))

### LINKED ISSUE #1936: Remove auth from redux
let's put auth in atmosphere instead of redux.

acceptance criteria:
- [ ] remove redux persist storage
- [ ] put auth token directly in local storage (or optionally a cookie)
- [ ] store auth token object as a local object in relay

effort: 11 points

### LINKED ISSUE #2005: redirect from /signin to /me if logged in
There was a regression when moving to the new login module where i can be logged in & still end up on the /signin page. This will be fixed alongside #1799

Acceptance Critieria
- redirect from /signin to /me if logged in

Effort: 10.5

### PR DESCRIPTION
**Pull request by mattkrick**
Removed over 1000 LOCs and 3 dependencies. Possibly my finest PR.
Fixes #1799 fixes #1936 fixes #2005

TEST
- [ ] can login
- [ ] can logout
- [ ] can accept invitation via email
- [ ] going to /admin/graphql as a normie gets you redirected to home
- [ ] going to `/` when you're logged in gets you redirected to `/me`
- [ ] creating a new account redirects you to /welcome
- [ ] going to /welcome after completing step 2 gets you redirected to /me
- [ ] segment and raven still include user details

### CODE REVIEW

**Comment by jordanh** [comment]
Only the finest PRs for us! Testing now...

**Comment by jordanh** [comment]
# Test Cycle 1
## Tests

- [ ] can login
- [ ] can logout
- [ ] can accept invitation via email
- [ ] going to /admin/graphql as a normie gets you redirected to home
- [ ] going to `/` when you're logged in gets you redirected to `/me`
- [ ] creating a new account redirects you to /welcome
- [ ] going to /welcome after completing step 2 gets you redirected to /me
- [ ] segment and raven still include user details

File: webpack/webpack.config.dll.js [hunk 1]
```diff
@@ -34,8 +33,6 @@ module.exports = {
       'redux-form',
       'redux-raven-middleware',
       'redux-segment',
-      'redux-storage-engine-localstorage',
```
Comment (jordanh) [inline webpack/webpack.config.dll.js]: These lines needed to be removed too in order for `build:dll` to succeed.

File: src/client/client.js [hunk 2]
```diff
@@ -6,36 +6,38 @@ import Root from './Root';
...
-})();
+if (__PRODUCTION__) {
+  StyleSheet.rehydrate(window.__APHRODITE__);
+}
+render(
+  <Root store={store} />,
+  document.getElementById('root')
+);
+
+/*
```
Comment (jordanh) [inline src/client/client.js]: Is this going to cramp @ackernaut's style?

File: src/universal/components/Action/Action.js [hunk 3]
```diff
@@ -60,7 +49,7 @@ const Action = () => {
         <AsyncRoute isPrivate path="/welcome" mod={welcome} />
         <AsyncRoute path="/admin/graphql" mod={graphql} />
         <AsyncRoute path="/admin/impersonate/:newUserId" mod={impersonate} />
-        <AsyncRoute path="/invitation/:id" mod={invitation} />
+        <AsyncRoute isPrivate path="/invitation/:inviteToken" mod={invitation} />
```
Comment (jordanh) [inline src/universal/components/Action/Action.js]: Is this right? Should this be isPrivate?

File: src/universal/components/AnalyticsIdentifier.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
+// @flow
+/*
+ * Currently, we only use this in the dashboard views, not the meeting views. That's probably OK for now, but not perfect.
+ */
+import PropTypes from 'prop-types';
```
Comment (jordanh) [inline src/universal/components/AnalyticsIdentifier.js]: This is fun!

File: src/universal/redux/authDuck.js [hunk 5]
```diff
@@ -1,117 +0,0 @@
-import jwtDecode from 'jwt-decode';
```
Comment (jordanh) [inline src/universal/redux/authDuck.js]: 👋

**Comment by jordanh** [comment]
Test cycle 1 complete! Any idea why email invites might be failing?

File: src/universal/components/Action/Action.js (hunk 3 above)
Comment (mattkrick) [reply src/universal/components/Action/Action.js]: yeah, the way new auth was designed, all the redirects happen in the same window. Before, when you weren't logged in, you'd see a modal asking you to log in. Now, the only way to log in is via the /signup route, so i just redirect there.

**Comment by mattkrick** [comment]
man, invites were passing when i wrote this, lemme dig in...

**Comment by mattkrick** [comment]
well that's embarrassing. fixed!

**Comment by jordanh** [comment]
# Test Cycle 1
## Tests

- [x] can login
- [x] can logout
- [x] can accept invitation via email
- [x] going to /admin/graphql as a normie gets you redirected to home
- [x] going to `/` when you're logged in gets you redirected to `/me`
- [x] creating a new account redirects you to /welcome
- [ ] ❌going to /welcome after completing step 2 gets you redirected to /me
- [ ] segment and raven still include user details

## Test Notes

## going to /welcome after completing step 2 gets you redirected to /me

This take me to the welcome wizard instead, then an exception on trying to complete step 2.

**Comment by jordanh** [comment]
# Test Cycle 3
## Tests

- [x] can login
- [x] can logout
- [x] can accept invitation via email
- [x] going to /admin/graphql as a normie gets you redirected to home
- [x] going to /admin/graphql as `su` gets you into graphiql
- [x] going to `/` when you're logged in gets you redirected to `/me`
- [x] creating a new account redirects you to /welcome
- [ ] going to /welcome after completing step 2 gets you redirected to /me
- [x] segment and raven still include user details

## Test Notes

## going to /welcome after completing step 2 gets you redirected to /me

This take me to the welcome wizard instead, then an exception on trying to complete step 2.

**Comment by jordanh** [comment]
Ok! All tests passed! This is GTG

**Pull request merged**